adds saga pattern, custom retry policies, defining custom errors (#1)
temporalio/edu-errstrat-typescript-code: README.md

# edu-errors-typescript-code

This repository provides code used for exercises and demonstrations
included in the TypeScript version of the

> [Crafting an Error Handling Strategy with TypeScript](https://learn.temporal.io/courses/tbd)

Reviewer: Just leaving a comment here to remind us to fix this URL once we have it

training course.

> It's important to remember that the example code used in this course was designed to support learning a specific aspect of Temporal, not to serve as a ready-to-use template for implementing a production system.

Reviewer 2: ```suggestion
The example code used in this course is designed to support learning a specific aspect of Temporal.
It's not meant to be used as a ready-to-use template for implementing production systems.
```

> For the exercises, make sure to run `temporal server start-dev --ui-port 8080 --db-filename clusterdata.db` in one terminal to start the Temporal server. For more details on this command, please refer to the `Setting up a Local Development Environment` chapter in the course. Note: If you're using the Gitpod environment to run this exercise, you can skip this step.

Reviewer 2: Consider breaking these down into bulleted prerequisites.

* Complete [Setting up a Local Development Environment](...) chapter
* Unless you're using Gitpod, ensure the Temporal CLI server is running on port 8080:

  ```
  temporal server start-dev \
    --ui-port 8080 \
    --db-filename temporal.db <-- notice how this does not use "cluster" or "data"
  ```

etc

## Hands-On Exercises

| Directory Name                   | Exercise                                               |
| :------------------------------- | :----------------------------------------------------- |
| `exercises/handling-errors`      | [Exercise 1](exercises/handling-errors/README.md)      |
| `exercises/non-retryable-errors` | [Exercise 2](exercises/non-retryable-errors/README.md) |
| `exercises/rollback-with-saga`   | [Exercise 3](exercises/rollback-with-saga/README.md)   |

> ## Reference

Reviewer 2: Consider "References" here, which is what these are.
